firmware/hw_layer/rtc_helper.cpp

```diff
@@ -4,23 +4,56 @@
  *
  * @date Feb 5, 2014
  * @author Andrey Belomutskiy, (c) 2012-2020
+ * @author Nathan Schulte, (c) 2022
  */
 
 #include "pch.h"
 
-#include <string.h> 
+#include <string.h>
 #include <time.h>
 
 #include "rtc_helper.h"
+
+#if EFI_PROD_CODE
 #include <sys/time.h>
 
+// Lua needs this function, but we don't necessarily have to implement it
+extern "C" int _gettimeofday(timeval* tv, void* tzvp) {
+	(void)tv; (void)tzvp;
+	return 0;
+}
+#endif // EFI_PROD_CODE
+
+#if EFI_RTC
+const char * ISO_UTC_DATE_TIME_FORMAT = "%04u-%02u-%02uT%02u:%02u:%02uZ";
+
+void convertRtcDateTimeToIsoUtc(RTCDateTime * timespec, char * isoUtcDateTime) {
+	uint32_t second = timespec->millisecond / 1000;
+	uint16_t minute = second / 60;
+	second -= minute * 60;
```

Comment:
There may be a way to reduce cycle count on this computation if it's a concern ("floor(div 60)", [floored] mult 60), but I haven't figured out what that looks like.

Reply:
These never run in a high priority place, so we don't really care about cost, I don't think. All math ops are fast other than 64 bit integer divide, which is software emulated.